client/src/main/java/com/paypal/selion/platform/grid/LocalSelendroidNode.java

```diff
@@ -15,191 +15,168 @@
+            throw e;
         }
-        String host = "localhost";
-        String hubPort = Config.getConfigProperty(ConfigProperty.SELENIUM_PORT);
-        String registrationUrl = String.format("http://%s:%s/grid/register", host, hubPort);
 
-        try {
-            int port = new LocalGridConfigFileParser().getPort() + 2;
-            startSelendroidDriverNode(port);
-            waitForNodeToComeUp(port,
-                    "Encountered problems when attempting to register the Selendroid Node to the local Grid");
-            isRunning = true;
-            logger.log(Level.INFO, "Attached SelendroidDriver node to local hub " + registrationUrl);
-        } catch (Exception e) {
-            logger.log(Level.SEVERE, e.getMessage(), e);
-            throw new GridException("Failed to start a local Selendroid Node", e);
+        if (instance == null) {
+            getLocalServerComponent();
         }
+        super.boot(testSession);
+        LOGGER.exiting();
     }
 
     @Override
     public void shutdown() {
-        if (server != null) {
-            try {
-                server.stopSelendroid();
-                logger.log(Level.INFO, "Local Selendroid Node has been stopped");
-            } catch (Exception e) {
-                String errorMsg = "An error occured while attempting to shut down the local Selendroid Node. Root cause: ";
-                logger.log(Level.SEVERE, errorMsg, e);
-            }
+        LOGGER.entering();
+        if (instance == null) {
+            LOGGER.exiting();
+            return;
         }
-
+        super.shutdown();
+        LOGGER.exiting();
     }
 
-    private void startSelendroidDriverNode(int port) throws Exception {
-        logger.entering(port);
-        List<String> args = new ArrayList<String>();
-        args.add("-deviceScreenshot");
-        args.add("-logLevel");
-        args.add("VERBOSE");
-        args.add("-hub");
-        args.add("http://127.0.0.1:" + Config.getIntConfigProperty(ConfigProperty.SELENIUM_PORT) + "/grid/register");
-        args.add("-port");
-        args.add(Integer.toString(port));
-        args.add("-selendroidServerPort");
-        args.add(Integer.toString(Config.getIntConfigProperty(ConfigProperty.SELENDROID_SERVER_PORT)));
-
-        // Specify the aut folder so its contents will be monitored and read by Selendroid.
-        String autFolder = Config.getConfigProperty(ConfigProperty.MOBILE_APP_FOLDER);
-        if ((autFolder != null) && (!autFolder.trim().isEmpty())) {
-            args.add("-folder");
-            args.add(autFolder);
+    private void validateConfiguredOptions() {
+        // Make sure the configured internal selendroid server port is not already in use
+        int selendroidServerPort = Config.getIntConfigProperty(ConfigProperty.SELENDROID_SERVER_PORT);
+        if (PortProber.pollPort(selendroidServerPort)) {
+            throw new IllegalArgumentException("Specified SELENDROID_SERVER_PORT is already in use.");
```

Comment:
We could do that as well. But I have seen a lot of people not being able to understand the `cant't bind to port` error message that Java throws. So it would be a good value addition if we provide a good error message.